From bb279eea97ef39d853604dcf2bfb1f1a745baf22 Mon Sep 17 00:00:00 2001
From: Allen Wild <allenwild93@gmail.com>
Date: Sat, 1 Feb 2020 13:20:37 -0500
Subject: [PATCH] Revert "kbuild: Fail if gold linker is detected"

It's true that the gold linker can't link the kernel, but Yocto scripts
run "make" a bunch of times during configure and dtb steps and don't
always override LD, resulting in these steps (sometimes silently)
failing.

Yocto does make sure the BFD linker is used when actually building
vmlinux and modules itself, so revert this error to allow all those
other tasks to succeed.

This reverts commit 75959d44f9dc8e44410667009724e4e238515502.
---
 scripts/Kconfig.include | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/scripts/Kconfig.include b/scripts/Kconfig.include
index a5fe72c504ff..40cdb465285e 100644
--- a/scripts/Kconfig.include
+++ b/scripts/Kconfig.include
@@ -39,9 +39,6 @@ as-instr = $(success,printf "%b\n" "$(1)" | $(CC) $(CLANG_FLAGS) -c -x assembler
 $(error-if,$(failure,command -v $(CC)),compiler '$(CC)' not found)
 $(error-if,$(failure,command -v $(LD)),linker '$(LD)' not found)
 
-# Fail if the linker is gold as it's not capable of linking the kernel proper
-$(error-if,$(success, $(LD) -v | grep -q gold), gold linker '$(LD)' not supported)
-
 # machine bit flags
 #  $(m32-flag): -m32 if the compiler supports it, or an empty string otherwise.
 #  $(m64-flag): -m64 if the compiler supports it, or an empty string otherwise.
-- 
2.32.0

